app/main.py

```diff
@@ -1,3 +1,76 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+import math
+import datetime
+from typing import Union, Tuple
+
+from app.car import Car
+from app.customer import Customer
+from app.shop import Shop
+
+
+def shop_trip() -> None:
+    with open("app/config.json", "r") as file:
+        config_file = json.load(file)
+
+    fuel_price = config_file["FUEL_PRICE"]
+    customers = [
+        Customer(
+            customer["name"],
+            customer["product_cart"],
+            customer["location"],
+            customer["money"],
+            Car(
+                customer["car"]["brand"],
+                customer["car"]["fuel_consumption"],
+            ),
+        )
+        for customer in config_file["customers"]
+    ]
+    shops = [
+        Shop(
+            shop["name"],
+            shop["location"],
+            shop["products"],
+        )
+        for shop in config_file["shops"]
+    ]
+
+    for customer in customers:
+        print(f"{customer.name} has {customer.money} dollars")
+        cheapest_prices: Tuple[Union[Shop | None], float | None] = (None, None)
+        for shop in shops:
+            shop_dist = math.dist(customer.location, shop.location)
+            cost_of_get_shop = (shop_dist * fuel_price
+                                * customer.car.fuel_consumption / 100)
+            shop_products_cost = sum(
+                quantity * shop.products[product]
+                for product, quantity in customer.product_cart.items()
+            )
+            final_price = round(cost_of_get_shop * 2 + shop_products_cost, 2)
+            print(f"{customer.name}'s trip to the "
+                  f"{shop.name} costs {final_price}")
+            if cheapest_prices[1] is None or cheapest_prices[1] > final_price:
+                cheapest_prices = (shop, final_price)
+
+        if cheapest_prices[1] <= customer.money:
```

Comment:
If shops will be empty

Reply:
Yes, it can be None. But there is no information in task what to do in that case... Theoretically it's not for the case "{customer.name} doesn't have enough money..."